### DISCUSSION TITLE
Files Droped on Widget Becoming Draggable

### DISCUSSION THREAD

**Discussion by plutotom** [johnlindquist/kit#1075] [Q&A] [answered]
I have been playing around with the idea of using the `drop` future and then taking that dropped file and displaying it in a small 'dock' like box that is always on top. This is helpful for temporary hold files while I move around my PC looking for the location I want to place my file.

I have noticed that the `path` panel has files that are draggable, so I played with the idea of copying the file that has been dropped to a temp directory then calling the path and setting the start path to that temp directory. This somewhat worked but I would love to use a widget so that I can edit the look and add further interactions to the widget. 

Been loving Script-Kit and am super excited to see where it goes.

**Comment by johnlindquist** [comment]
@plutotom 

I worked on this today. I'll get it published in the *next* release (maybe tomorrow?) 😇

https://user-images.githubusercontent.com/36073/215907862-e98d7c09-d5db-443e-b92e-0e7004097286.mp4

**Reply by Ambushfall** [reply]
Amazing drag n drop!

🏆

**Comment by plutotom** [comment, marked as answer]
Here is the code @johnlindquist shared on discord that will be possible in the update 1.47.0. 

If anyone makes anything cool with this drop-it-blow, I would love to see more about how we can manipulate this feature. 
### Drag n' Drop!
```import "@johnlindquist/kit"

let files = []

let w = await widget(
  `<div class="flex flex-col">
<h2 class="text-2xl">Drop Files</h2>
<span class="text-sm py-1 cursor-pointer" v-for="(file, index) in files" :key="file" :data-file="file">{{file}}</span>
</div>
`,
  {
    containerClass: `p-4 h-screen w-screen overflow-auto`,
    width: 300,
    height: 300,
    draggable: false,
    state: {
      files,
    },
  }
)

w.onDrop(event => {
  if (event?.dataset?.files) {
    files.push(...event.dataset.files)
    w.setState({
      files,
    })
  }
})

w.onMouseDown(event => {
  if (event.dataset.file) {
    startDrag(event.dataset.file)
  }
})
